app/src/main/java/com/duckduckgo/app/browser/DuckDuckGoWebView.kt

```diff
@@ -267,12 +274,55 @@ class DuckDuckGoWebView : WebView, NestedScrollingChild3 {
         }
     }
 
+    private suspend fun isWebMessageListenerSupported(
+        dispatchers: DispatcherProvider,
+        webViewVersionProvider: WebViewVersionProvider,
+    ): Boolean {
+        return withContext(dispatchers.io()) {
+            webViewVersionProvider.getFullVersion()
+                .compareSemanticVersion(WEB_MESSAGE_LISTENER_WEBVIEW_VERSION)?.let { it >= 0 } ?: false
+        } && WebViewFeature.isFeatureSupported(WebViewFeature.WEB_MESSAGE_LISTENER)
+    }
+
+    @SuppressLint("RequiresFeature", "AddWebMessageListenerUsage")
+    suspend fun safeAddWebMessageListener(
```

Comment:
We need to offer up some way for the caller to know when this fails. For some features, silently not adding the web message listener is fine, but for others we'll need to take action (show error message in the UI etc..).

Even returning a `Boolean` indicating if it was successfully added would suffice.

Reply:
Good idea! I’ll add that

Reply:
@CDRussell I’ve added the boolean here: https://github.com/duckduckgo/Android/pull/4676/commits/84e71abab78ba762ecc25f207f0e6f0c6d44deec